### PR TITLE
League statistics second refactor

### PR DESCRIPTION
**Pull request by Lithnotep**
Add last refactor for league statistics complete with all methods and tests.

### CODE REVIEW

File: lib/league_statistics.rb [hunk 1]
```diff
@@ -40,118 +35,94 @@ def goals_scored_by_team
...
+    team_average_goals_home = Hash.new(0)
     @game_collection.each do |game|
       goals_and_games_by_team_home[game.home_team_id][0] += game.home_goals.to_i
       goals_and_games_by_team_home[game.home_team_id][1] += 1
       end
     goals_and_games_by_team_home.each do |team|
-    team_average_goals_home[team[0]] = average_goals(team[1][0], team[1][1])
-    team_average_goals_home
+    team_average_goals_home[team[0]] = average(team[1][0], team[1][1].to_f)
       end
```
Comment (takeller) [inline lib/league_statistics.rb]: End needs to be properly indented

File: lib/league_statistics.rb [hunk 2]
```diff
@@ -40,118 +35,94 @@ def goals_scored_by_team
...
-    end
-    max_average = team_average_goals_home.max_by{|team| team[1]}
-    highest_scoring_home_team = @teams_collection.find do |team|
+  def best_offense_team_id
+    max_average = average_goals_by_team.max_by{|team| team[1]}
+    best_offense = @teams_collection.find do |team|
       team.id == max_average[0]
-    end
-    highest_scoring_home_team.team_name
+      end
```
Comment (takeller) [inline lib/league_statistics.rb]: This one looks like it is off too.

File: lib/league_statistics.rb [hunk 3]
```diff
@@ -40,118 +35,94 @@ def goals_scored_by_team
...
-    end
-    min_average = team_average_goals_home.min_by{|team| team[1]}
-    lowest_scoring_home_team = @teams_collection.find do |team|
+  def worst_offense_team_id
+    min_average = average_goals_by_team.min_by{|team| team[1]}
+    worst_offense = @teams_collection.find do |team|
       team.id == min_average[0]
-    end
-    lowest_scoring_home_team.team_name
+      end
```
Comment (takeller) [inline lib/league_statistics.rb]: And here

**Review by takeller — CHANGES_REQUESTED** [review]

File: lib/league_statistics.rb [hunk 4]
```diff
@@ -1,24 +1,19 @@
 require_relative './statistics'
+require_relative './mathable'
 
 class LeagueStatistics < Statistics
+  include Mathable
+  attr_reader :game_collection, :game_teams_collection, :teams_collection
```
Comment (sagemlee) [inline lib/league_statistics.rb]: You shouldnt need these attr readers here

**Review by sagemlee — APPROVED** [review]

**Review by takeller — APPROVED** [review]

**Pull request merged**